src/main/java/com/fittracker/fittracker/entity/Product.java

```diff
@@ -0,0 +1,189 @@
+package com.fittracker.fittracker.entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+import org.hibernate.annotations.CreationTimestamp;
+
+@Entity
+@Table(name = "products", schema = "fittracker")
+@IdClass(Product.ProductId.class)
+public class Product {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+    @Id
+    private int version;
+    private String name;
+    private int kcal;
+    private int carbs;
+    private int protein;
+    private int fat;
+    private UUID userId;
+
+    @CreationTimestamp
```

Comment:
we want the timestamp to be generated by postgres, so we could do `@Column(insertable = false)` instead (so hibernate won't attempt to insert null)